### ISSUE TITLE
[Bug] Set-Party Trainers can Spawn with Duplicates of their Signature Pokemon

### ISSUE THREAD

**Issue by Snailman11** [pagefaultgames/pokerogue#4893]
### Describe the bug

https://github.com/pagefaultgames/pokerogue/blob/ae8efeedf8ca520f2beb17ca8093586de5775ca0/src/data/trainer-config.ts#L1323C4-L1323C5
Lacey can Spawn with two Primarinas

-Set Trainers Duplicate check does not check basic evolutions accurately when looking at Signature/Guaranteed Pokémon

-Set Trainers Duplicate check looks at the group of Signature Pokémon in the random group which prevents those Pokémon from spawning even if the Pokémon isn't chosen.

Wave Seed: 3wDsT9xudeWKK14ovL5XXm6n 0
game-data.ts:1024 Seed: 3wDsT9xudeWKK14ovL5XXm6n
battle-scene.ts:1512 Wave Seed: íıþĭĎóĲįĞğđąąëîĩİĆïĒĒħðĨ 186

### Reproduction

Load the session data provided or rig the seed using overrides, set STARTING_WAVE to just before a fight with a Trainer with Signature Pokemon, then see that the game rolls a duplicate.

### Expected behavior

Set Trainers with preferred Pokemon: Gym Leaders, Elite Four, Champions, Rivals, Evil Team Admins and Leaders, should not be able to spawn with a Party that contains duplicate Pokémon unless specifically intended by SignatureSpecies.

### Screenshots / Videos

![Screenshot (448)](https://github.com/user-attachments/assets/8f39790b-587f-4af0-adce-96907e904d7e)

https://github.com/user-attachments/assets/819b2cb2-2c1d-4341-878e-6d0323e58032

### Session export file

[LaceyDoublePrimaWaveBefore.txt](https://github.com/user-attachments/files/17788789/LaceyDoublePrimaWaveBefore.txt)
-Should be the wave before

[Damo's Experience.txt](https://github.com/user-attachments/files/17788837/Damo.s.Experience.txt)
-Damo's Experience

### Additional context

PR that fixed the overall issue had some oversights ([Bug] Preventing Duplicate Species in Trainer Battles)
https://github.com/pagefaultgames/pokerogue/pull/4368

**_Important Insight by Moka_**
https://discord.com/channels/1125469663833370665/1176874654015684739/1299555916730863709
> This was added but there are problems left with it so it's only half working as intended 
The issue is that the fixed mons of the trainers can still create duplicates in some cases (thanks for reminding me about this..)
The thing though is that from my understanding that oversight is good in a way because if it worked as Intended it would block some Pokemon that didn't need to be blocked from spawning in some other cases. So that PR needs to get reworked 

**Whether a Set Trainer's SignatureSpecies are their evolved forms or not seems inconsistent **
https://github.com/pagefaultgames/pokerogue/blob/ae8efeedf8ca520f2beb17ca8093586de5775ca0/src/data/trainer-config.ts#L1323C4-L1323C5

> some fixed pools have randomness to them so excluding everything means some mons get excluded even if they don't end up getting used from the fixed pool 

-Set Trainers have a set team, and then a chance to spawn with a defined group of specific Pokémon.
 - If the duplicates check sees a Pokémon in that group, then it blacklists that Pokémon
 - this causes problems because a blacklisted Pokémon might not be the one that actually is chosen from that group. 
and it becomes impossible for that Pokémon to spawn for that Trainer if it is not chosen from the defined group.

> TLDR:  fixed pool for Hassle has Dragalge, not it's non evolved species, and your code checks for the non evolved form, so does not see  Dragalge as duplicate
The issue with fixed pools actually being random is why I left it at that because the fix is not as simple if we want to address both issues

**Comment by Snailman11** [comment]
Link to a Discord Bug report with Bryony and Liepard
(Team Flare Admin has 2 Liepard (Static trainer Party))
https://discord.com/channels/1125469663833370665/1320427835805925447
